### PR TITLE
feat(lua): Add `getSourceValue`

### PR DESCRIPTION
**Pull request by jfrickmann**
Implementation of what is requested in issue #781.

### CODE REVIEW

File: radio/src/lua/api_general.cpp [hunk 1]
```diff
@@ -2420,9 +2550,11 @@ const luaL_Reg opentxLib[] = {
   { "getGlobalTimer", luaGetGlobalTimer },
   { "getRotEncSpeed", luaGetRotEncSpeed },
   { "getValue", luaGetValue },
+  { "getSourceValue", luaGetSourceValue },
   { "getRAS", luaGetRAS },
   { "getTxGPS", luaGetTxGPS },
   { "getFieldInfo", luaGetFieldInfo },
+  { "getSourceInfo", luaGetFieldInfo },
```
Comment (raphaelcoeffic) [inline radio/src/lua/api_general.cpp:2557]: @jfrickmann Is the duplication intentional? Is that just for convenience of API?

Comment (jfrickmann) [reply radio/src/lua/api_general.cpp:2557]: Yes, it is what Robert requested in pt. 2 here: https://github.com/EdgeTX/edgetx/issues/781#issuecomment-1193895564

**Comment by pfeerick** [comment]
@CapnBry Since you have some spare time now that ELRS v3 is in the review mirror 🤣 ... could you kick the tires on this and see what you think.

**Comment by CapnBry** [comment]
Yeah I will take a look when I have a chance. 3.0 isn't exactly in the rearview mirror just yet as the number of "this doesn't work at all any more!" issues have exploded

I did look over the code, I just haven't altered anything to use it yet to verify.

**Comment by pfeerick** [comment]
lol... you mean to say I shouldn't have just updated *all* of my gear over to 3.0 yesterday afternoon? :-P Just if you get a chance in the next 7-10 days - want at least one Lua developer to say "good enough for 2.8" before pressing that tempting merge button. 😁

**Comment by CapnBry** [comment]
You bet! I am sorry it is taking so long to have a look at it, but I will have a chance Monday or Tuesday for sure.

**Comment by CapnBry** [comment]
Alright, everything works except for one thing, which is that the `getSourceValue()` returns true for the second parameter so long as `TELEMETRY_STREAMING`. However, this should only return true if the sensor value is current, which is separate from telemetry on the whole, i.e. `telemetryItems[qr.quot].isFresh()` for those type of item.

Tested
-- Case sensitive only match
-- getSourceInfo returns nil if not found, or table desc/unit/id/name
-- getSourceInfo works with both ID and string
-- getSourceValue works with both ID and string
-- getSourceValue returns nil before item has a value, allowing a default to be presented to the user
-- getSourceValue returns false for second parameter before item has a value
-- getSourceValue returns value when value is present
-- getSourceValue returns true for second parameter when telemetry streaming
-- getSourceValue returns previous value after telemetry lost
-- getSourceValue returns previous value after sensor lost
-- getSourceValue returns false for second parameter when telemetry lost
-- getSourceValue returns true for second parameter when telemetry recovered

-- getSourceValue returns false for second parameter if sensor lost: **FAIL**

One interesting thing I found while testing is that the telemetry items lose their value (reset to returning nil) after a USB Mass Storage connection. I am sure this is intentional but thought I'd bring it up just in case it was not.

**Comment by jfrickmann** [comment]
Something was messed up with the git branch, so I started a new one on #2316.

I have made the change requested above - please check!

**Comment by pfeerick** [comment]
Closing due to #2316 ... don't you just love `git` at times!!

**Pull request closed without merging**